### PR TITLE
Added ConcurrentTestRunner

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added ConcurrentTestRunner class
2. Added ConcurrentTest model

### CODE REVIEW

File: src/foam/nanos/bench/BenchmarkRunner.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          latch.countDown();
+        }
+      });
+      // start the thread
+      thread.start();
+    }
+
+    try {
+      // wait until latch reaches 0, or until timeout is reached
+      if (timeout_ != 0) {
```
Comment (kgrgreer) [inline src/foam/nanos/bench/BenchmarkRunner.java:131]: spacing

File: src/foam/nanos/bench/BenchmarkRunner.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    // set up the test
+    test_.setup(x);
+
+    // get start time
+    long startTime = System.currentTimeMillis();
+
+    // execute all the threads
+    for ( Thread thread : threads ) {
+      thread = new Thread(new Runnable() {
```
Comment (kgrgreer) [inline src/foam/nanos/bench/BenchmarkRunner.java:115]: Create a ThreadGroup for the test and create all threads in that group.
Provide a toString() for the threads.
Both of these will make it easier when looking at thread dumps.

File: src/foam/nanos/bench/BenchmarkRunner.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        latch.await(timeout_, TimeUnit.MILLISECONDS);
+      } else {
+        latch.await();
+      }
+    } catch (Throwable ignored) {}
+
+    // calculate length taken
+    long endTime = System.currentTimeMillis();
+    long duration = endTime - startTime;
+    System.out.println(threadCount_ +
```
Comment (kgrgreer) [inline src/foam/nanos/bench/BenchmarkRunner.java:141]: I would like to have a compound TPS and transactions per thread.

File: src/foam/nanos/bench/BenchmarkRunner.java (hunk 2 above)
Comment (kirkeaton) [reply src/foam/nanos/bench/BenchmarkRunner.java:115]: What should the toString for the threads return?

**Pull request merged**